### PR TITLE
Parameterization: Make Iterative_authalic_parameterization faster (WIP)

### LINKED ISSUE #6694: Parallelization with SMP: Iterative Authalic Parameterization
Hello!
I'd like to know if it's possible to benefit from parallelization when running the Iterative_authalic_parameterizer_3 algorithm. I'm running it on a huge mesh (59M faces, 2.4G size) and it's taking a very long time (weeks), even on an HPC environment. Memory is not an issue, but running single-thread is. I have TBB support compiled-in, but I see no way of telling the algorithm to use it.
Any pointer in this direction would be appreciated.
Thanks!

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

I hard coded the copy of a sparse matrix.   We have to add this properly to a Solver API.
I removed debug output that writes the mesh after each iteration.

## Release Management

* Affected package(s): Surface_mesh_parameterization
* Issue(s) solved (if any): fix #6694

### CODE REVIEW

**Comment by afabri** [comment]
This PR requires that we add copying a matrix as operation, unless we think that matrices are `Assignable`.  Any opinion on this @sloriot ?

**Comment by sloriot** [comment]
We can either update the concept or add a new one. Provided that Eigen is the only current model we have, checking if Eigen is fine with this and updating the concept is fine to me. MKL being in oneAPI now, we might want to re-add it as a model at some point.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-30

**Comment by lrineau** [comment]
This PR is still a draft. What should be done before it can be marked as ready?

**Comment by afabri** [comment]
> This PR is still a draft. What should be done before it can be marked as ready?

We have to discuss if it makes sense to add to some solver package concepts that the matrix type is assignable.   @soesau can you  please take over as you are the maintainer of the solver package.

**Comment by soesau** [comment]
I added an assignment operator to the concept SparseLinearAlgebraTraits_d. It seems that besides Surface_mesh_parameterization_3 also the package Heat_method_3 rely on the assignment operator of CGAL::Eigen_solver_traits.

There is another concept SparseLinearAlgebraWithFactorTraits_d which also uses Vector and Matrix types without specifying them. I guess we should add them both to the concept and including the assignment operator?

**Comment by sloriot** [comment]
`SparseLinearAlgebraWithFactorTraits_d` refines `SparseLinearAlgebraTraits_d` so they are somehow "inherited"

File: Solver_interface/doc/Solver_interface/Concepts/SparseLinearAlgebraTraits_d.h [hunk 1]
```diff
@@ -221,6 +221,10 @@ operator*(const NT& c, const Matrix& M);
 friend Matrix
 operator+(const Matrix& M0, const Matrix& M1);
 
+/// Assignment operator
```
Comment (afabri) [inline Solver_interface/doc/Solver_interface/Concepts/SparseLinearAlgebraTraits_d.h:224]: I though we only wanted to write that the concept ` Matrix`  refines ` Assignable`.

Comment (soesau) [reply Solver_interface/doc/Solver_interface/Concepts/SparseLinearAlgebraTraits_d.h:224]: I see, makes sense.
And the assignment operator alone is actually not sufficient as `Assignable` refines `CopyAssignable` and `CopyConstructible`.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-36.

@afabri you approve the changes?

**Comment by afabri** [comment]
As it is my PR I cannot approve.

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Iterative_authalic_parameterizer_3.h [hunk 2]
```diff
@@ -50,7 +50,7 @@
 #include <iomanip>
 #include <unordered_set>
 
-#define DEBUG_L0 1 // @fixme
+#define DEBUG_L0 0 // @fixme
```
Comment (lrineau) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Iterative_authalic_parameterizer_3.h:53]: That macro name should be prefixed by `CGAL_`.

**Pull request merged**